feat: Added instrumentation support for iovalkey

mbackonja opened the pull request:
## Description

Add subscriber-based instrumentation for **iovalkey**, a Valkey client that is a friendly fork of ioredis. This enables database metrics and tracing for users migrating from Redis to Valkey.

Key changes:
- Add `IovalkeySubscriber` extending `DbOperationSubscriber` with diagnostics channel integration
- Configure diagnostics channels to hook into iovalkey's `sendCommand` method
- Add comprehensive test suite covering both CommonJS and ESM module systems
- Support iovalkey versions >=0.1.0
- Use Valkey-specific metric naming (e.g., `Datastore/operation/Valkey/set`) to distinguish from Redis traffic
- Add `valkey_host` and `valkey_port` test parameters with fallback to redis parameters for compatibility

Implementation follows the existing ioredis instrumentation pattern since iovalkey is a fork of ioredis and maintains the same internal structure.

## How to Test

Automated tests have been added and pass:
```bash
# Run iovalkey CommonJS tests
EXTERNAL_MODE=none SKIP_C8=true ./bin/run-versioned-tests.sh iovalkey

# Run iovalkey ESM tests
EXTERNAL_MODE=none SKIP_C8=true ./bin/run-versioned-tests.sh iovalkey-esm

# Run linter
npm run lint
```

Test coverage includes:
- Metrics generation (Datastore/Valkey/*)
- Segment creation in transaction traces
- Instance attributes (host, port, database_name, key)
- Configuration flags (instance_reporting, database_name_reporting)
- Database selection tracking (SELECT command)
- Error handling
- Promise chains
- Pipeline operations
- Transaction lifecycle edge cases

Tested against iovalkey versions: 0.1.0, 0.2.2, 0.3.3

## Related Issues

Closes #3508

## Notes

Question for maintainers: I noticed the dashboard tracking system in `dashboards/constants.js` and `dashboards/library-usage.json` hasn't been updated since June 2024 and is missing recently added packages. Should this system be maintained going forward? If so, I can update it to include `iovalkey` in this PR.

bizob2828 commented on test/versioned/iovalkey-esm/iovalkey.test.js:
```diff
@@ -0,0 +1,217 @@
+// Indicates unique database in Valkey. 0-15 supported.
+const DB_INDEX = 5
+
+test('iovalkey instrumentation', async (t) => {
```
you can delete all these tests. we just created ioredis-esm tests for various reasons that are not applicable here.

bizob2828 commented on test/versioned/iovalkey/iovalkey.test.js:
```diff
@@ -0,0 +1,342 @@
+const { removeModules } = require('../../lib/cache-buster')
+
+// Indicates unique database in Valkey. 0-15 supported.
+const DB_INDEX = 3
```
our team can clean these up but this index is being used elsewhere.  can you change this value to `4`

mbackonja commented:
**Changes made:**
- Refactored `IoRedisSubscriber` to accept optional `packageName` and `system` parameters
- Simplified `IovalkeySubscriber` to inherit from `IoRedisSubscriber`
- Removed unnecessary diagnostics channel configurations (kept only `built/Redis.js`)
- Deleted ESM test directory
- Updated `DB_INDEX` to `4` and `SELECTED_DB` to `9`
- Changed `minAgentVersion` to `13.9.0`

**Testing:**
- All iovalkey tests pass (3 versions: 0.1.0, 0.2.2, 0.3.3)
- All ioredis tests pass (10 versions)
- Linter is clean

Ready for another review when you have time @bizob2828
